File: Python/Array/BestTimeToBuyAndSellStock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        buy = float('inf')
        res = 0
        currMax = 0
        for price in prices:
            if price < buy:
                buy = price
            else:
                currMax = max(currMax, price - buy)
                res = max(currMax, res)

        return res

File: Java/Array/BestTimeToBuyAndSellStock.java
class Solution {
    public int maxProfit(int[] prices) {
        int max = 0;
        int buy = prices[0];
        for (int price : prices) {
            int profit = price - buy;
            max = Math.max(max, profit);
            if (profit < 0) {
                buy = price;
            }
        }

        return max >= 0 ? max : 0;
    }
}